File: Tcm.Domain/Model/SchoolEducationSubCourse.cs
using Framework.Persistence.Ef;
using System.Collections.Generic;

namespace Tcm.Domain.Model
{
    public class SchoolEducationSubCourse : EntityBase<long>
    {
        public int SchoolId { get; set; }
        public School School { get; set; }
        public short EducationSubCourseId { get; set; }
        public EducationSubCourse EducationSubCourse { get; set; }
        public short MajorId { get; set; }
        public Major Major { get; set; }
        public short ClassCount { get; set; }
        //public long StudentId { get; set; }
        //public Student Student { get; set; }

        public List<ClassRoom> ClassRooms { get; set; }
    }
}

File: Tcm.Interface.WebUi/src/app/pages/school/models/SchoolEducationSubCourse.ts
export class SchoolEducationSubCourse {
    Id: Number;
    SchoolId: Number;
    EducationSubCourseId: Number;
    MajorId: number;
    EducationSubCourseName?: string;

    constructor(educationSubCourseId, schoolId,majorId){
        this.EducationSubCourseId = educationSubCourseId;
        this.SchoolId = schoolId;
        this.MajorId = majorId;
    }
}
